src/main/java/org/terasology/projsndwv/systems/ExtractorSystem.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.projsndwv.systems;
+
+import org.terasology.engine.Time;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.common.DisplayNameComponent;
+import org.terasology.logic.delay.DelayManager;
+import org.terasology.logic.delay.DelayedActionTriggeredEvent;
+import org.terasology.logic.inventory.InventoryComponent;
+import org.terasology.logic.inventory.InventoryManager;
+import org.terasology.logic.inventory.events.BeforeItemPutInInventory;
+import org.terasology.logic.inventory.events.InventorySlotChangedEvent;
+import org.terasology.projsndwv.TempBeeRegistry;
+import org.terasology.projsndwv.components.BeeComponent;
+import org.terasology.projsndwv.components.ExtractorComponent;
+import org.terasology.projsndwv.components.LocusSampleComponent;
+import org.terasology.projsndwv.components.ProcessingComponent;
+import org.terasology.projsndwv.genetics.components.GeneticsComponent;
+import org.terasology.registry.In;
+import org.terasology.utilities.random.MersenneRandom;
+
+@RegisterSystem(RegisterMode.ALWAYS) // TODO: Authority
+public class ExtractorSystem extends BaseComponentSystem {
+    @In
+    private DelayManager delayManager;
+
+    @In
+    private Time time;
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private InventoryManager inventoryManager;
+
+    /**
+     * The slot index for the input slot of the extractor.
```

Comment:
Another trivial bit and probably more personal preference than anything, but for fields (no chance of stuff like param or return tags) I like the look of one liner javadoc, at least when the descriptive text is relatively short. Helps condense into a nice clean list of fields. Up to you tho!